lib/active_merchant/billing/gateways/stripe.rb

```diff
@@ -90,14 +90,24 @@ def purchase(money, payment, options = {})
           return Response.new(false, direct_bank_error)
         end
 
+        if options[:payment_to_confirm]
+          r = commit(:post, "payment_intents/#{options[:payment_to_confirm]}/confirm", {}, options)
+          return r
+        end
+
         MultiResponse.run do |r|
           if payment.is_a?(ApplePayPaymentToken)
             r.process { tokenize_apple_pay_token(payment) }
             payment = StripePaymentToken.new(r.params["token"]) if r.success?
           end
           r.process do
             post = create_post_for_auth_or_purchase(money, payment, options)
-            commit(:post, 'charges', post, options)
+
+            if options[:three_d_secure]
```

Comment:
what about:
```
api_endpoint = options[:three_d_secure] ? 'payment_intents' : 'charges'
commit(:post, api_endpoint, post, options)
```

Reply:
good point 👍 and actually we can do just:
```
commit(:post, options[:three_d_secure] ? 'payment_intents' : 'charges', post, options)
```